helm-chart/renku-notebooks/templates/network-policies.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "notebooks.fullname" . }}
+  labels:
+    app: {{ template "notebooks.name" . }}
+    chart: {{ template "notebooks.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  podSelector:
+    matchLabels:
+      component: "singleuser-server"
+      app.kubernetes.io/component: {{ .Values.amalthea.crdNames.kind | lower  }}
+  policyTypes:
+    - Ingress
+  ingress:
+    - from:
```

Comment:
I know I think that it could be that amalthea itself or the ui-server or perhaps some other services may be interested in calling this. That is why I left it open like this.

Reply:
But I dont mind closing it up now and then opening it later if we need it. 

Keep in mind from inside a user session you still cannot access another session's git sidecar rpc server at all because the egress network policy in amalthea prohibits egress pointed at anything inside the cluster.

Reply:
It is actually better to be more restrictive than less. So I made the network policy as you suggest. Only the notebook service can access the rpc server.